### PR TITLE
ISPN-10567 Allow container exposed ports when CONTAINER is the server driver

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.jboss.org/browse/ISPN-10567

### CODE REVIEW

**Comment by diegolovison** [comment]
The first Jenkins run must match master.
The second Jenkins will change `org.infinispan.test.server.container.preferContainerExposedPorts=true`. The Jenkins result must match.

**Comment by diegolovison** [comment]
https://ci.infinispan.org/job/Infinispan/job/PR-7273/1/testReport/ = 26,647 tests
https://ci.infinispan.org/job/Infinispan/job/PR-7273/2/testReport/ = 26,647 tests
@tristantarrant it is ready for review.

**Comment by tristantarrant** [comment]
I have made some changes in https://github.com/infinispan/infinispan/pull/7429/files#diff-4dc1891beeaf3a8a95499198219cff0eR190 which will change the way you implement this. Wait for that PR to be merged before rebasing.

**Comment by diegolovison** [comment]
@tristantarrant ready for review

**Comment by tristantarrant** [comment]
@diegolovison I don't see the code that checks for the org.infinispan.test.server.container.preferContainerExposedPorts property

**Comment by diegolovison** [comment]
Added it back. I will run the test with and without enabling the property. Once done, I will let you know. tks

**Comment by diegolovison** [comment]
`org.infinispan.server.persistence.PersistenceIT` still failing on master. Let's wait for https://issues.jboss.org/browse/ISPN-10879 before merging this PR

File: server/runtime/src/test/java/org/infinispan/server/test/ContainerInfinispanServerDriver.java [hunk 1]
```diff
@@ -50,6 +52,7 @@
    CountdownLatchLoggingConsumer latch;
    ImageFromDockerfile image;
    private File rootDir;
+   private final boolean preferContainerExposedPorts = Boolean.valueOf(System.getProperty("org.infinispan.test.server.container.preferContainerExposedPorts", "false"));
```
Comment (tristantarrant) [inline server/runtime/src/test/java/org/infinispan/server/test/ContainerInfinispanServerDriver.java]: You can replace this with Boolean.getBoolean("org.infinispan.test.server.container.preferContainerExposedPorts")

**Comment by tristantarrant** [comment]
Merged, thanks

**Comment by alanfx** [comment]
:beers:

**Pull request merged**